keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+1.24:
+- disable by default
+- allow gate to enable the feature
+- release note
+
+1.26:
+- enable by default
+- allow gate to disable the feature
+- release note
+
+1.28:
+- remove gate
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: AdminNetworkPolicy
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature by itself has no effect on the cluster.
+Creating a AdminNetworkPolicy does have an effect on
+the cluster, however they must be specifically created, which means the
+administrator is aware of the impact.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Once enabled, the feature can be disabled via feature gate. However, disabling
+the feature may cause created cluster-scoped policy resources to be deleted,
+which may impact the security of the cluster. Administrators must make provision
+to secure their cluster by other means before disabling the feature gate.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+n/a
+
+###### Are there any tests for feature enablement/disablement?
+
+n/a
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+While focusing only on E/W traffic for now prevents functional overloading of the 
+AdminNetworkPolicy API, it also creates some concerns around scalability.  Specifically, 
+the implementations will need to keep track of all in cluster IPs in order to only 
+select cluster-internal entities while ignoring everything else.  Specific graduation 
+requirements have been constructed to ensure this does not become an issue for 
+future versions and implementations. 
+
+###### Will enabling / using this feature result in any new API calls?
+
+Enabling this feature by itself will have no impact on the cluster and no
+new API calls will be made.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+Enabling this feature will introduce new API types as described in the [design](#design-details)
+section. The supported number of objects per cluster will depend on the individual
+CNI providers who will be responsible to provide the implementation to realize
+these resources.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+n/a
```

Comment:
Isn't the answer to this implementation specific?

Reply:
Yep, added a note

Reply:
done